File: src/main/scala/top/criwits/scaleda/idea/ScaledaBundle.scala
package top.criwits.scaleda
package idea

import com.intellij.DynamicBundle
import org.jetbrains.annotations.{Nls, NonNls, NotNull, PropertyKey}
import top.criwits.scaleda.idea.ScaledaBundle.bundle

final class ScaledaBundle extends DynamicBundle(bundle)

object ScaledaBundle {
  @NonNls
  private val bundle = "bundles.ScaledaBundle"
  private val instance = new ScaledaBundle

  @Nls
  def message(@NotNull @PropertyKey(resourceBundle = "bundles.ScaledaBundle") key: String, @NotNull params: Any*): String =
    instance.getMessage(key, params)
}

File: src/main/scala/top/criwits/scaleda/idea/ScaledaBundle.java
package top.criwits.scaleda.idea;

import com.intellij.DynamicBundle;
import org.jetbrains.annotations.*;

import java.util.Locale;
import java.util.function.Supplier;

public final class ScaledaBundle extends DynamicBundle {
    @NonNls private static final String BUNDLE = "bundles.ScaledaBundle";
    // TODO: support i18n
    // note that if we replace INSTANCE with BUNDLE_ZH on launch, shown language may change
    @NonNls private static final String BUNDLE_ZH = "bundles.ScaledaBundle_zh";
    private static final ScaledaBundle INSTANCE = new ScaledaBundle();

    private ScaledaBundle() { super(BUNDLE); }

    @NotNull
    public static @Nls String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key, @NotNull Object ...params) {
        return INSTANCE.getMessage(key, params);
    }

}

